### PR TITLE
Node 18 Upgrade + Knex Improvements

### PR DESCRIPTION
**Pull request by MikeVerdicchio**
- Upgrade to Node18 runtime
- Adjust knex pool and timeout settings
- Add index on trace_id to improve assumed user queries
- Remove unused code and format

### CODE REVIEW

File: migrations/20231108104112_trace_id_index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function (knex) {
+  return knex.raw(
+    "CREATE INDEX IF NOT EXISTS trace_id_index ON query USING btree (trace_id)"
```
Comment (giovtorres) [inline migrations/20231108104112_trace_id_index.js:7]: 👍🏼

File: src/lib/helpers.js [hunk 2]
```diff
@@ -25,18 +25,6 @@ function getProxiedBody(clusterBody, proxyId, proxyHost) {
   return newBody;
 }
 
-function getUsernameFromAuthorizationHeader(header) {
```
Comment (giovtorres) [inline src/lib/helpers.js:28]: not used?

File: src/lib/knex.js [hunk 3]
```diff
@@ -6,4 +6,13 @@ if (!connectionString) {
...
   connection: connectionString,
+  /** It is recommended to set min to 0 so idle connections can be terminated */
+  /** @see https://knexjs.org/guide/#pool */
+  pool: {
+    min: 0,
+    max: 10,
+  },
+  /** How long knex waits before throwing a timeout error while acquiring a connection */
+  /** @see https://knexjs.org/guide/#acquireconnectiontimeout */
+  acquireConnectionTimeout: 10000,
```
Comment (giovtorres) [inline src/lib/knex.js:17]: Should this be an env var with a default?

File: src/lib/helpers.js (hunk 2 above)
Comment (MikeVerdicchio) [reply src/lib/helpers.js:28]: Not used - we switched from using a UI endpoint (which required auth) to an API one (that doesn't)

File: src/lib/knex.js [hunk 4]
```diff
@@ -6,4 +6,13 @@ if (!connectionString) {
 exports.knex = require("knex")({
   client: "pg",
   connection: connectionString,
+  /** It is recommended to set min to 0 so idle connections can be terminated */
+  /** @see https://knexjs.org/guide/#pool */
+  pool: {
+    min: 0,
+    max: 10,
```
[the remark this answers is not in the document]
Comment (MikeVerdicchio) [reply src/lib/knex.js]: Based on some research, it looks like there might be some stats emitted from the underlying lib. I might follow up with another PR to see if I can add that cleanly.

Great call with the envar approach - added!

File: src/lib/knex.js (hunk 3 above)
Comment (MikeVerdicchio) [reply src/lib/knex.js]: Added!

**Review by giovtorres — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**